src/librustc/lint/internal.rs

```diff
@@ -7,8 +7,9 @@ use crate::lint::{
 };
 use errors::Applicability;
 use rustc_data_structures::fx::FxHashMap;
-use syntax::ast::Ident;
+use syntax::ast::{Ident, Item, ItemKind};
 use syntax::symbol::{sym, Symbol};
+use syntax_pos::ExpnInfo;
 
 declare_lint! {
     pub DEFAULT_HASH_TYPES,
```

Comment:
Should I also rename internal lints and use
```rust
declare_tool_lint! {
    pub rustc::DEFAULT_HASH_TYPES,
    ...
}
```
?

Reply:
jop

Reply:
I just tried to rename this and hit this error:
```
Building stage0 compiler artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
   Compiling rustc_data_structures v0.0.0 (/home/pkrones/Documents/work/rust/src/librustc_data_structures)
error[E0710]: an unknown tool name found in scoped lint: `rustc::default_hash_types`
  --> src/librustc_data_structures/lib.rs:30:32
   |
30 | #![cfg_attr(not(stage0), allow(rustc::default_hash_types))]
   |                                ^^^^^

error: aborting due to previous error
```

I mean, that the error is displayed in stage0 is expected. But with `cfg_attr(not(stage0), _)` this attribute shouldn't even exist when reaching the error handling code:
https://github.com/rust-lang/rust/blob/a96ba969156d257e5d5b692946fa8fe40ed6543a/src/librustc/lint/levels.rs#L284-L290